components/ws-manager-api/typescript/src/client-provider-source.ts

```diff
@@ -14,17 +14,17 @@ export interface WorkspaceManagerClientProviderSource {
     getAllWorkspaceClusters(): Promise<WorkspaceClusterWoTLS[]>;
 }
 
-
 @injectable()
 export class WorkspaceManagerClientProviderEnvSource implements WorkspaceManagerClientProviderSource {
     protected _clusters: WorkspaceCluster[] | undefined = undefined;
+    readonly applicationCluster = process.env.GITPOD_INSTALLATION_SHORTNAME ?? "";
```

Comment:
Instead of reading from the env across the onject hiearchy let's trx to consume from `config.ts`. I think `Configuration.installation` is what we want.

Reply:
This is a good point. I'm not comfortable with reading the environment here either. I'm trying a different DI-based approach at the moment where the applicationCluster is becomes part of `WorkspaceManagerClientProvider` and is set up in the DI container.

Branch is here: https://github.com/gitpod-io/gitpod/compare/af/di